### PR TITLE
renaming pico module names and api package names

### PR DESCRIPTION
**Pull request by trentjeff**
Addresses: https://github.com/helidon-io/helidon/issues/6553

### CODE REVIEW

File: pico/api/src/test/java/io/helidon/pico/api/DefaultPicoConfigTest.java [hunk 1]
```diff
@@ -14,12 +14,16 @@
...
  */
 
-package io.helidon.pico;
+package io.helidon.pico.api;
 
 import java.util.Map;
 
 import io.helidon.common.config.Config;
 import io.helidon.config.ConfigSources;
+import io.helidon.pico.api.DefaultBootstrap;
```
Comment (tomas-langer) [inline pico/api/src/test/java/io/helidon/pico/api/DefaultPicoConfigTest.java:23]: These imports are not required, probably added as part of moving the packages.
You can run `optimize imports` in Idea that only does it for GIT modified files, which helps with this kind of problems

File: pico/api/src/test/java/io/helidon/pico/api/PriorityAndServiceTypeComparatorTest.java [hunk 2]
```diff
@@ -14,17 +14,17 @@
...
 import java.util.ArrayList;
 import java.util.Comparator;
 import java.util.List;
 
 import io.helidon.common.Weighted;
 import io.helidon.common.Weights;
-import io.helidon.pico.testsubjects.PicoServices1Provider;
-import io.helidon.pico.testsubjects.PicoServices2Provider;
-import io.helidon.pico.testsubjects.PicoServices3Provider;
+import io.helidon.pico.api.testsubjects.PicoServices1Provider;
```
Comment (tomas-langer) [inline pico/api/src/test/java/io/helidon/pico/api/PriorityAndServiceTypeComparatorTest.java:25]: Please fix here (and in any other class that has the same problem)

File: pico/maven-plugin/src/main/java/io/helidon/pico/maven/plugin/ApplicationCreatorMojo.java [hunk 3]
```diff
@@ -35,13 +35,13 @@
...
     /**
-     * The classname to use for the Pico {@link io.helidon.pico.Application} class.
+     * The classname to use for the Pico {@link io.helidon.pico.api.Application} class.
      * If not found the classname will be inferred.
      */
     @Parameter(property = PicoServicesConfig.NAME + ".application.class.name", readonly = true
                // note: the default value handling doesn't work here for "$$"!!
-//               defaultValue = DefaultApplicationCreator.APPLICATION_NAME
-    )
+               //               defaultValue = DefaultApplicationCreator.APPLICATION_NAME
```
Comment (tomas-langer) [inline pico/maven-plugin/src/main/java/io/helidon/pico/maven/plugin/ApplicationCreatorMojo.java:43]: this reformatting does not look good, was that intentional?

File: pico/runtime/src/main/java/io/helidon/pico/runtime/Dependencies.java [hunk 4]
```diff
@@ -24,21 +24,22 @@
...
+import io.helidon.pico.api.DependencyInfo;
+import io.helidon.pico.api.ElementInfo;
+import io.helidon.pico.api.InjectionPointInfo;
+import io.helidon.pico.api.QualifierAndValue;
+import io.helidon.pico.api.ServiceInfoCriteria;
 
 /**
  * This is the class the code-generator will target that will be used at runtime for a service provider to build up its
- * dependencies expressed as {@link io.helidon.pico.DependenciesInfo}.
+ * dependencies expressed as {@link DependenciesInfo}.
```
Comment (tomas-langer) [inline pico/runtime/src/main/java/io/helidon/pico/runtime/Dependencies.java:42]: We should use fully qualified links in javadoc, so if there is only a javadoc reference, there is no import in class

File: pico/tests/resources-pico/src/main/java/io/helidon/pico/tests/pico/interceptor/XImpl.java [hunk 5]
```diff
@@ -30,7 +30,7 @@
 import jakarta.inject.Singleton;
 
 /**
- * This test case is applying {@link InterceptorBasedAnno} (an {@link io.helidon.pico.InterceptedTrigger}) using the no-arg
+ * This test case is applying {@link InterceptorBasedAnno} (an {@link io.helidon.pico.pi.InterceptedTrigger}) using the no-arg
```
Comment (tomas-langer) [inline pico/tests/resources-pico/src/main/java/io/helidon/pico/tests/pico/interceptor/XImpl.java:33]: typo - `.pi.` instead of `.api.`

File: pico/tests/resources-pico/src/test/java/io/helidon/pico/tests/pico/JavaxTest.java [hunk 6]
```diff
@@ -62,7 +62,7 @@ void tearDown() {
     void applicationScopeToSingletonScopeTranslation() {
         ServiceProvider<AnApplicationScopedService> sp = services.lookupFirst(AnApplicationScopedService.class);
         assertThat(sp.toString(),
-                   equalTo("AnApplicationScopedService:INIT"));
+                                 equalTo("AnApplicationScopedService:INIT"));
```
Comment (tomas-langer) [inline pico/tests/resources-pico/src/test/java/io/helidon/pico/tests/pico/JavaxTest.java:65]: this should not be changed

File: pico/tools/src/main/java/io/helidon/pico/tools/JavaC.java [hunk 7]
```diff
@@ -231,7 +232,8 @@ public boolean hasWarnings() {
 
         public ToolsException maybeGenerateError() {
             if (!isSuccessful()) {
-                return new ToolsException("Creator compilation error");
+                return new ToolsException("Compilation error encountered:\n"
```
Comment (tomas-langer) [inline pico/tools/src/main/java/io/helidon/pico/tools/JavaC.java:235]: This is an unrelated change, should not be part of this PR at all

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: pico/api/src/test/java/io/helidon/pico/api/DefaultPicoConfigTest.java (hunk 1 above)
Comment (trentjeff) [reply pico/api/src/test/java/io/helidon/pico/api/DefaultPicoConfigTest.java:23]: I just ran optimize imports now - it didn't find anything additional, but perhaps because It is not comparing against the main branch... will push shortly.

File: pico/maven-plugin/src/main/java/io/helidon/pico/maven/plugin/ApplicationCreatorMojo.java (hunk 3 above)
Comment (trentjeff) [reply pico/maven-plugin/src/main/java/io/helidon/pico/maven/plugin/ApplicationCreatorMojo.java:43]: no, this was something I guess IDEA did also.

File: pico/api/src/test/java/io/helidon/pico/api/PriorityAndServiceTypeComparatorTest.java (hunk 2 above)
Comment (trentjeff) [reply pico/api/src/test/java/io/helidon/pico/api/PriorityAndServiceTypeComparatorTest.java:25]: this is expected as the test packages changed as well.

File: pico/maven-plugin/src/main/java/io/helidon/pico/maven/plugin/ApplicationCreatorMojo.java (hunk 3 above)
Comment (trentjeff) [reply pico/maven-plugin/src/main/java/io/helidon/pico/maven/plugin/ApplicationCreatorMojo.java:43]: next push

File: pico/runtime/src/main/java/io/helidon/pico/runtime/Dependencies.java (hunk 4 above)
Comment (trentjeff) [reply pico/runtime/src/main/java/io/helidon/pico/runtime/Dependencies.java:42]: next push

File: pico/tests/resources-pico/src/main/java/io/helidon/pico/tests/pico/interceptor/XImpl.java (hunk 5 above)
Comment (trentjeff) [reply pico/tests/resources-pico/src/main/java/io/helidon/pico/tests/pico/interceptor/XImpl.java:33]: interesting how checkstyles / spotbugs didn't pick that up.

File: pico/tests/resources-pico/src/test/java/io/helidon/pico/tests/pico/JavaxTest.java (hunk 6 above)
Comment (trentjeff) [reply pico/tests/resources-pico/src/test/java/io/helidon/pico/tests/pico/JavaxTest.java:65]: next push

File: pico/api/src/test/java/io/helidon/pico/api/DefaultPicoConfigTest.java (hunk 1 above)
Comment (trentjeff) [reply pico/api/src/test/java/io/helidon/pico/api/DefaultPicoConfigTest.java]: pushed

File: pico/maven-plugin/src/main/java/io/helidon/pico/maven/plugin/ApplicationCreatorMojo.java (hunk 3 above)
Comment (trentjeff) [reply pico/maven-plugin/src/main/java/io/helidon/pico/maven/plugin/ApplicationCreatorMojo.java]: pushed

File: pico/runtime/src/main/java/io/helidon/pico/runtime/Dependencies.java (hunk 4 above)
Comment (trentjeff) [reply pico/runtime/src/main/java/io/helidon/pico/runtime/Dependencies.java]: pushed, as well as a whole bunch more in this category

File: pico/tests/resources-pico/src/main/java/io/helidon/pico/tests/pico/interceptor/XImpl.java (hunk 5 above)
Comment (trentjeff) [reply pico/tests/resources-pico/src/main/java/io/helidon/pico/tests/pico/interceptor/XImpl.java]: pushed

File: pico/tests/resources-pico/src/test/java/io/helidon/pico/tests/pico/JavaxTest.java (hunk 6 above)
Comment (trentjeff) [reply pico/tests/resources-pico/src/test/java/io/helidon/pico/tests/pico/JavaxTest.java]: pushed

File: pico/tools/src/main/java/io/helidon/pico/tools/JavaC.java (hunk 7 above)
Comment (trentjeff) [reply pico/tools/src/main/java/io/helidon/pico/tools/JavaC.java:235]: Discussed offline and agreed to keep this in.

**Review by barchetta — APPROVED** [review]
